gulpfile.js

```diff
@@ -297,6 +302,218 @@ gulp.task('minifyRelease', ['generateStubs'], function() {
     });
 });
 
+gulp.task('deploy', function(done) {
+    var cacheControl = "max-age=3600";
+    var concurrencyLimit = 2000;
+
+    var argv = yargs.usage('Usage: delpoy -b [Bucket Name] -d [Upload Directory]')
+        .demand(['b', 'd']).argv;
+
+    var uploadDir = argv.d;
+    var bucketName = argv.b;
+
+    var s3 = new aws.S3({
+        maxRetries: 10,
+        retryDelayOptions: {
+            base: 500
+        }
+    });
+
+    var existingBlobs = [];
+    var totalFiles = 0;
+    var uploaded = 0;
+
+    // get all files currently in bucket
+    function listAll(callback, marker) {
```

Comment:
Rather than make functions inside the task, declare them ourtide of the tasks but with more descriptive names.  We might eventually break this master gulpfile.js up into small modules for easier maintenance, but that can be done in a future PR.